pkg/core/hostid/aws.go

```diff
@@ -1,51 +1,40 @@
 package hostid
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/signalfx/signalfx-agent/pkg/utils/timeutil"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
 	log "github.com/sirupsen/logrus"
+
+	"github.com/signalfx/signalfx-agent/pkg/utils/timeutil"
 )
 
 // AWSUniqueID constructs the unique EC2 instance of the underlying host.  If
 // not running on EC2, returns the empty string.
 func AWSUniqueID(cloudMetadataTimeout timeutil.Duration) string {
-	c := http.Client{
+	// Pass in the HTTP client so cloudMetadataTimeout from the config
+	// is respected.
+	c := &http.Client{
 		Timeout: cloudMetadataTimeout.AsDuration(),
 	}
 
-	resp, err := c.Get("http://169.254.169.254/2014-11-05/dynamic/instance-identity/document")
+	sess, err := session.NewSession(aws.NewConfig().WithHTTPClient(c))
 	if err != nil {
 		log.WithFields(log.Fields{
 			"detail": err,
-		}).Info("No AWS metadata server detected, assuming not on EC2")
+		}).Info("Failed to create new session for AWS metadata collection")
 		return ""
 	}
-	defer resp.Body.Close()
 
-	body, err := ioutil.ReadAll(resp.Body)
+	client := ec2metadata.New(sess)
+	doc, err := client.GetInstanceIdentityDocument()
```

Comment:
The SDK has this method https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/#EC2Metadata.Available (and with context version) that we can use to avoid polluting the logs if the agent is not on EC2.
Shall we take advantage of it?

Reply:
Also, does this throw an error `session.NewSession` if you are not on EC2? if yes, then I don't think there is a point to execute `Available`

Reply:
That's a good point, will update to use the `Available` method. Right now, `session.NewSession` does not throw an error, if it's not an EC2 instance.

Reply:
Updated.